src/java/picard/vcf/filter/FilterVcf.java

```diff
@@ -87,14 +103,26 @@ public static void main(final String[] args) {
     }
 
     @Override
-    protected int doWork() {
+    public int doWork() {
         IOUtil.assertFileIsReadable(INPUT);
         IOUtil.assertFileIsWritable(OUTPUT);
 
-        final List<VariantFilter>  variantFilters = CollectionUtil.makeList(new AlleleBalanceFilter(MIN_AB), new FisherStrandFilter(MAX_FS), new QdFilter(MIN_QD));
-        final List<GenotypeFilter> genotypeFilters = CollectionUtil.makeList(new GenotypeQualityFilter(MIN_GQ), new DepthFilter(MIN_DP));
         final VCFFileReader in = new VCFFileReader(INPUT, false);
-        final FilterApplyingVariantIterator iterator = new FilterApplyingVariantIterator(in.iterator(), variantFilters, genotypeFilters);
+        final List<VariantFilter>  variantFilters = new ArrayList<VariantFilter>(4);
+        variantFilters.add(new AlleleBalanceFilter(MIN_AB));
+        variantFilters.add(new FisherStrandFilter(MAX_FS));
+        variantFilters.add(new QdFilter(MIN_QD));
+        if( JAVASCRIPT_FILE != null) {
+            try {
+                variantFilters.add(new VariantContextJavascriptFilter(JAVASCRIPT_FILE, in.getFileHeader()));
+            } catch(IOException error) {
```

Comment:
error can be final.  (you'll notice you get a lot of these comments from reviewers.  It's because we all use intellij which autodetects what can be made final and pops up a little yellow bar next to it if it's not)